### PR TITLE
fix: check for above-zero target height increase before setting isActive

### PR DESCRIPTION
**Pull request by MaciejBaj**
### Summary by OpenAI

**New Features:**
- Added checks for above-zero target height increase in `pallets/xdns/src/lib.rs`.
- Introduced new test case `test_determines_none_for_two_non_increasing_elements` in `primitives/src/lib.rs`.
- Modified the `DisabledValidators` type in `runtime/t0rn-parachain/src/parachain_config.rs`.
- Updated import statements in `node/parachain/src/rpc.rs`, `node/standalone/src/rpc.rs`.
- Implemented new functions and methods related to side effects and transactions in `pallets/circuit/src/lib.rs`, `pallets/circuit/src/machine/test.rs`, `primitives/src/circuit/traits.rs`.
- Added a new function `fetch_all_active_xtx` to the `PortalRuntimeApi` trait and its implementations across multiple files.
- Removed a call to `XDNS::process_overview(System::block_number())` in `runtime/mini-mock/src/lib.rs`.
- Enabled serialization and deserialization for the `SideEffect` struct in `types/src/sfx.rs`.

> 🎉 Code changes are here, making things clear, 🚀
> New features abound, improvements all around. 🌟
> With each PR merge, we surge, 💪
> On the path of innovation, we continue our journey without reservation. 🛤️

### CODE REVIEW

File: runtime/t0rn-parachain/src/parachain_config.rs [hunk 1]
```diff
@@ -107,7 +107,7 @@ impl pallet_session::Config for Runtime {
 impl pallet_aura::Config for Runtime {
     type AllowMultipleBlocksPerSlot = ConstBool<false>;
     type AuthorityId = AuraId;
-    type DisabledValidators = ();
+    type DisabledValidators = Session;
```
Comment (3h4x) [inline runtime/t0rn-parachain/src/parachain_config.rs:110]: > A way to check whether a given validator is disabled and should not be authoring blocks. Blocks authored by a disabled validator will lead to a panic as part of this module’s initialization.

https://docs.rs/pallet-aura/latest/pallet_aura/pallet/trait.Config.html#associatedtype.DisabledValidators

So what will happen when we will switch it to `Session`?

Comment (MaciejBaj) [reply runtime/t0rn-parachain/src/parachain_config.rs:110]: Session exposes interface to check disabled validators. currently however the list of disabledValidators is empty on t0rn so we'd need to pin slashing rules that would ask Session::disableIndex

**Review by gvko — CHANGES_REQUESTED** [review]
minor stuff.
Also, build doesn't suceed.

**Review by gvko — APPROVED** [review]
build & test passing, all good

**Pull request merged**